doc/application/application.rst

```diff
@@ -3,129 +3,478 @@
+
+You will need an ELF binary image for debugging purposes.  The build system
+generates the image in the build directory.  By default, the kernel binary name
+is :file:`zephyr.elf`. The name can be changed using a Kconfig option.
+
+We will use the standard 1234 TCP port to open a :abbr:`GDB (GNU Debugger)`
+server instance. This port number can be changed for a port that best suits the
+development environment.
+
+You can run Qemu to listen for a "gdb connection" before it starts executing any
+code to debug it.
+
+.. code-block:: bash
+
+   qemu -s -S <image>
+
+will setup Qemu to listen on port 1234 and wait for a GDB connection to it.
 
-* :makevar:`ZEPHYR_BASE`: Sets the path to the kernel's base
-  directory.  This variable is usually set by the
-  :file:`zephyr_env.sh` script.  It can be used to get the kernel's
-  base directory, as used in the boilerplate above, or it can be
-  overridden to select an specific instance of the kernel.
+The options used above have the following meaning:
 
-* :makevar:`BOARD`: Selects the board that the application's
-  build will use for the default configuration.
+* ``-S`` Do not start CPU at startup; rather, you must type 'c' in the
+  monitor.
+* ``-s`` Shorthand for :literal:`-gdb tcp::1234`: open a GDB server on
+  TCP port 1234.
 
-* :makevar:`CONF_FILE`: Indicates the name of a configuration fragment file.
-  This file includes the kconfig configuration values that override the
-  default configuration values.
+To debug with QEMU and to start a GDB server and wait for a remote connect, run
+the following inside the build directory of an application:
+
+.. code-block:: bash
+
+   ninja debugserver
+
+The build system will start a QEMU instance with the CPU halted at startup
+and with a GDB server instance listening at the TCP port 1234.
+
+Using a local GDB configuration :file:`.gdbinit` can help initialize your GDB
+instance on every run.
+In this example, the initialization file points to the GDB server instance.
+It configures a connection to a remote target at the local host on the TCP
+port 1234. The initialization sets the kernel's root directory as a
+reference.
+
+The :file:`.gdbinit` file contains the following lines:
+
+.. code-block:: bash
+
+   target remote localhost:1234
+   dir ZEPHYR_BASE
+
+.. note::
+
+   Substitute ZEPHYR_BASE for the current kernel's root directory.
+
+Execute the application to debug from the same directory that you chose for
+the :file:`gdbinit` file. The command can include the ``--tui`` option
+to enable the use of a terminal user interface. The following commands
+connects to the GDB server using :file:`gdb`. The command loads the symbol
+table from the elf binary file. In this example, the elf binary file name
+corresponds to :file:`zephyr.elf` file:
+
+.. code-block:: bash
+
+   $ gdb --tui zephyr.elf
+
+.. note::
+
+   The GDB version on the development system might not support the --tui
```

Comment:
This is a pre-existing issue that I don't have a quick fix for. Since I've just moved around the problematic documentation, I don't think this needs to be addressed to merge this series.

Reply:
if --tui then just use GDB as is?